### PR TITLE
fix(admin): reject AdminAPI call with empty tags

### LINKED ISSUE #13591: Error occurs on endpoints when `tags` parameter is empty
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

Kong 2.7.1

### Current Behavior

Requests with `?tags=` to the admin api results in a 500 error

```
curl -X GET 'http://localhost:8001/ca_certificates?tags='
```

Kong logs 

* `stringx.lua:32: argument 1 expected a 'string', got a 'userdata'`

```
2024/08/29 05:18:38 [error] 1335#0: *10353 [lua] api_helpers.lua:540: handle_error(): /usr/local/share/lua/5.1/lapis/application.lua:440: /usr/local/share/lua/5.1/pl/stringx.lua:32: argument 1 expected a 'string', got a 'userdata'
stack traceback:
   [C]: in function 'error'
   /usr/local/share/lua/5.1/pl/utils.lua:291: in function 'assert_arg'
   /usr/local/share/lua/5.1/pl/stringx.lua:32: in function 'assert_string'
   /usr/local/share/lua/5.1/pl/stringx.lua:194: in function 'split'
   /usr/local/share/lua/5.1/kong/api/endpoints.lua:166: in function 'extract_options'
   /usr/local/share/lua/5.1/kong/api/endpoints.lua:209: in function 'page_collection'
   /usr/local/share/lua/5.1/kong/api/endpoints.lua:319: in function 'fn'
   /usr/local/share/lua/5.1/kong/api/api_helpers.lua:311: in function </usr/local/share/lua/5.1/kong/api/api_helpers.lua:289>

stack traceback:
   [C]: in function 'error'
   /usr/local/share/lua/5.1/lapis/application.lua:440: in function 'handler'
   /usr/local/share/lua/5.1/lapis/application.lua:186: in function 'resolve'
   /usr/local/share/lua/5.1/lapis/application.lua:217: in function </usr/local/share/lua/5.1/lapis/application.lua:215>
   [C]: in function 'xpcall'
   /usr/local/share/lua/5.1/lapis/application.lua:215: in function 'dispatch'
   /usr/local/share/lua/5.1/lapis/nginx.lua:231: in function 'serve'
   /usr/local/share/lua/5.1/kong/init.lua:1867: in function 'admin_content'
   content_by_lua(nginx-kong.conf:333):2: in main chunk, client: 172.19.0.1, server: kong_admin, request: "GET /ca_certificates?tags= HTTP/1.1", host: "localhost:8001"
172.19.0.1 - - [29/Aug/2024:05:18:39 +0000] "GET /ca_certificates?tags= HTTP/1.1" 500 42 "-" "curl/7.81.0"
```

### Expected Behavior

The API should return a valid response, such as an empty list of CA certificates or a relevant error message indicating that the tags parameter is required, if applicable. The system should handle empty tags gracefully without causing an internal server error.

### Steps To Reproduce

run kong
run the curl `curl -X GET 'http://localhost:8001/ca_certificates?tags='`

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by nowNick**
### Summary

When querying entities that have tags assigned a user can send a request with `tags` parameter. Previously sending `tags=` (empty parameter) resulted in 500 error. This commit changes that so that it returns 400 error as empty explicit tags are not allowed.

### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [x] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com/pull/8099

### Issue reference

KAG-5496
Fix #13591

### CODE REVIEW

**Review by hutchic — APPROVED** [review]
LGTM. Does the openapi spec need to be updated to reflect?

File: changelog/unreleased/kong/fix-admin-api-for-empty-tags.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+message: Fix for querying admin API entities with emtpy tags
```
Comment (brentos) [inline changelog/unreleased/kong/fix-admin-api-for-empty-tags.yml:1]:
```suggestion
message: Fix for querying admin API entities with empty tags
```

File: spec/02-integration/04-admin_api/14-tags_spec.lua [hunk 2]
```diff
@@ -73,6 +73,16 @@ describe("Admin API - tags", function()
         end
       end)
 
+      it("filter by emtpy tag", function()
```
[the remark this answers is not in the document]
Comment (nowNick) [reply spec/02-integration/04-admin_api/14-tags_spec.lua]: Well apparently my left hand is faster than my right hand as I was writing this typo consistently 😅 
It's fixed, thanks! ✅

File: changelog/unreleased/kong/fix-admin-api-for-empty-tags.yml (hunk 1 above)
Comment (nowNick) [reply changelog/unreleased/kong/fix-admin-api-for-empty-tags.yml]: Fixed ✅

File: spec/02-integration/04-admin_api/03-consumers_routes_spec.lua [hunk 3]
```diff
@@ -282,6 +282,16 @@ describe("Admin API (#" .. strategy .. "): ", function()
...
       end)
+      it("returns bad request for empty tags", function()
+        local res = assert(client:send {
+          method = "GET",
+          path = "/consumers",
+          query = { tags = ngx.null}
+        })
+        res = assert.res_status(400, res)
+        local json = cjson.decode(res)
+        assert.same("invalid option (tags: tags cannot be null (or empty string))", json.message)
```
Comment (brentos) [inline spec/02-integration/04-admin_api/03-consumers_routes_spec.lua:293]: The error message seems a bit redundant -- what about `invalid option (tags: cannot be null or empty string)`
Removes nested parenthesis and duplicate "tags".

File: spec/02-integration/04-admin_api/14-tags_spec.lua [hunk 4]
```diff
@@ -73,6 +73,16 @@ describe("Admin API - tags", function()
         end
       end)
 
+      it("filter by empty tag", function()
```
Comment (brentos) [inline spec/02-integration/04-admin_api/14-tags_spec.lua:76]: Is there a way to test with empty string too?

**Comment by brentos** [comment]
Just a couple minor comments, otherwise LGTM

**Review by locao — APPROVED** [review]

File: spec/02-integration/04-admin_api/03-consumers_routes_spec.lua (hunk 3 above)
Comment (nowNick) [reply spec/02-integration/04-admin_api/03-consumers_routes_spec.lua]: ✅  I've improved the wording to match your suggestion.

File: spec/02-integration/04-admin_api/14-tags_spec.lua (hunk 4 above)
Comment (nowNick) [reply spec/02-integration/04-admin_api/14-tags_spec.lua:76]: Oh good catch. Turns out that sending explicit empty string is fine. I've added a test for that to make sure we don't return 500 response in that case.

**Review by brentos — APPROVED** [review]

**Pull request merged**